### PR TITLE
fix(react-native-usb): close all opened devices without exception

### LINKED ISSUE #13510: ConcurrentModificationException
openedConnections collection is iterated and modified concurently at the same time, we need to fix closeAllOpenedDevices and closeDevice methods to prevent this error. 

Sentry Issue: [SUITE-NATIVE-RQ](https://satoshilabs.sentry.io/issues/5615747913/?referrer=github_integration)

```
RuntimeException: Unable to pause activity {io.trezor.suite/io.trezor.suite.MainActivity}: java.lang.reflect.InvocationTargetException
    at android.app.ActivityThread.performPauseActivityIfNeeded(ActivityThread.java:4993)
    at android.app.ActivityThread.performPauseActivity(ActivityThread.java:4944)
    at android.app.ActivityThread.handlePauseActivity(ActivityThread.java:4896)
    at android.app.servertransaction.PauseActivityItem.execute(PauseActivityItem.java:46)
    at android.app.servertransaction.TransactionExecutor.executeLifecycleState(TransactionExecutor.java:176)
...
(8 additional frame(s) were not displayed)

InvocationTargetException: None
    at java.lang.reflect.Method.invoke(Method.java)
    at expo.modules.ReactActivityDelegateWrapper.invokeDelegateMethod(ReactActivityDelegateWrapper.kt:328)
    at expo.modules.ReactActivityDelegateWrapper.onPause(ReactActivityDelegateWrapper.kt:203)
    at com.facebook.react.ReactActivity.onPause(ReactActivity.java:53)
    at android.app.Activity.performPause(Activity.java:8150)
...
(14 additional frame(s) were not displayed)

ConcurrentModificationException: None
    at java.util.LinkedHashMap$LinkedHashIterator.nextNode(LinkedHashMap.java:757)
    at java.util.LinkedHashMap$LinkedEntryIterator.next(LinkedHashMap.java:790)
    at java.util.LinkedHashMap$LinkedEntryIterator.next(LinkedHashMap.java:788)
    at io.trezor.rnusb.ReactNativeUsbModule.closeAllOpenedDevices(ReactNativeUsbModule.kt:803)
    at io.trezor.rnusb.ReactNativeUsbModule.access$closeAllOpenedDevices(ReactNativeUsbModule.kt:35)
...
(32 additional frame(s) were not displayed)
```

### PR DESCRIPTION
**Pull request by yanascz**
## Related Issue

Resolve #13510

### CODE REVIEW

File: packages/react-native-usb/android/src/main/java/io/trezor/rnusb/ReactNativeUsbModule.kt [hunk 1]
```diff
@@ -234,15 +234,17 @@ class ReactNativeUsbModule : Module() {
 
     private fun closeDevice(deviceName: String) {
         Log.d(LOG_TAG, "Closing device $deviceName")
-        val usbConnection = getOpenedConnection(deviceName)
-        usbConnection.close()
+        getOpenedConnection(deviceName).close()
```
Comment (Nodonisko) [inline packages/react-native-usb/android/src/main/java/io/trezor/rnusb/ReactNativeUsbModule.kt:237]: Does this really solves the issue? It doesn't seems like anything that should affect functionallity.

Comment (yanascz) [reply packages/react-native-usb/android/src/main/java/io/trezor/rnusb/ReactNativeUsbModule.kt:237]: No, the important change is in `closeAllOpenedDevices()`. The variable here is just redundant.

File: packages/react-native-usb/android/src/main/java/io/trezor/rnusb/ReactNativeUsbModule.kt [hunk 2]
```diff
@@ -234,15 +234,17 @@ class ReactNativeUsbModule : Module() {
...
         openedConnections.remove(deviceName)
     }
 
     private fun closeAllOpenedDevices() {
         Log.d(LOG_TAG, "Closing all devices")
-        openedConnections.forEach { (deviceName, _) ->
-            closeDevice(deviceName)
+        with(openedConnections.iterator()) {
+            forEach {
+                it.value.close()
```
Comment (Nodonisko) [inline packages/react-native-usb/android/src/main/java/io/trezor/rnusb/ReactNativeUsbModule.kt:245]: Can you please explain little bit more why this is better than previous solution? Both seems pretty similar and I don't get it why one causes concurency issues and one not.

Comment (yanascz) [reply packages/react-native-usb/android/src/main/java/io/trezor/rnusb/ReactNativeUsbModule.kt:245]: In Java/Kotlin, it’s illegal to remove an entry from a map (or any other collection) while iterating over it using a for loop. One needs to use iterator or some alternative techniques, see https://www.baeldung.com/kotlin/map-remove-while-iterating for further details.

Comment (yanascz) [reply packages/react-native-usb/android/src/main/java/io/trezor/rnusb/ReactNativeUsbModule.kt:245]: The first commit contains alternative, more verbose, syntax. Lemme know if you prefer it more.

Comment (Nodonisko) [reply packages/react-native-usb/android/src/main/java/io/trezor/rnusb/ReactNativeUsbModule.kt:245]: Thanks for explanation now it makes more sense to me. Second version looks better IMHO.

**Review by Nodonisko — APPROVED** [review]

**Pull request merged**